src/code/z_kankyo.c

```diff
@@ -1,163 +1,992 @@
+}
+
+// Repurposed from OoT to be more general
+void Environment_ClearEventDayCount(void) {
+    gSaveContext.save.eventDayCount = 0;
+}
+
+// Repurposed from OoT to be more general
+u32 Environment_GetTotalDays(void) {
+    return gSaveContext.save.day;
+}
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_ForcePlaySequence.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_IsForcedSequenceDisabled.s")
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_PlayStormNatureAmbience.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD538.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_StopStormNatureAmbience.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_LerpAmbientColor.s")
+void Environment_WarpSongLeave(PlayState* play) {
+    gWeatherMode = WEATHER_MODE_CLEAR;
+    gSaveContext.save.cutsceneIndex = 0;
+    gSaveContext.respawnFlag = -3;
+    play->nextEntrance = gSaveContext.respawn[RESPAWN_MODE_RETURN].entrance;
+    play->transitionTrigger = TRANS_TRIGGER_START;
+    play->transitionType = TRANS_TYPE_FADE_WHITE;
+    gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_LerpDiffuseColor.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_SetupSkyboxStars.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_LerpFogColor.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSkyboxStar.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_LerpFog.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSkyboxStarsImpl.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/get_days_elapsed.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_Draw.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/reset_days_elapsed.s")
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/Environment_DrawSkyboxStars.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/get_current_day.s")
+void Environment_StopTime(void) {
+    sEnvIsTimeStopped = true;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD750.s")
+void Environment_StartTime(void) {
+    sEnvIsTimeStopped = false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD768.s")
+u8 Environment_IsTimeStopped(void) {
+    return sEnvIsTimeStopped;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD78C.s")
+u32 Environment_GetStormState(PlayState* play) {
+    u32 stormState = play->envCtx.stormState;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_800FD858.s")
+    if ((play->sceneId == SCENE_OMOYA) && (play->roomCtx.curRoom.num == 0)) {
+        stormState = ((gSaveContext.save.day >= 2) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_DEFENDED_AGAINST_THEM))
+                         ? STORM_STATE_ON
+                         : STORM_STATE_OFF;
+    }
```

Comment:
Wait some weather is determined by aliens?

Reply:
Yeah, the barn. If you lose than the roof is gone so would need to have the same storm state as outside.

Reply:
Oh that makes sense, I was assuming the storm was caused by the aliens or something but yeah, the roof being gone.